src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,137 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAY_LIST_CAPACITY = 10;
+    private T[] elements;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public ArrayList() {
+        this.elements = (T[]) new Object[DEFAULT_ARRAY_LIST_CAPACITY];
+        this.size = 0;
+    }
+
+    @SuppressWarnings("unchecked")
     @Override
     public void add(T value) {
-
+        if (size == elements.length) {
+            int newCapacity = (int) (elements.length * 1.5);
+            T[] newElements = (T[]) new Object[newCapacity];
+            System.arraycopy(elements, 0, newElements, 0, elements.length);
+            elements = newElements;
+        }
+        elements[size++] = value;
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Index can not be less then 0 and greater then size"
+            );
+        }
```

Comment:
Please, extract this check to a separate private method to avoid code duplication